internal/hcs/system.go

```diff
@@ -326,13 +336,113 @@ func (computeSystem *System) Properties(ctx context.Context, types ...schema1.Pr
 	return properties, nil
 }
 
+// statisticsInProc emulates what HCS does to grab statistics for a given container with a small
+// change to make grabbing the private working set total much more efficient.
+func (computeSystem *System) statisticsInProc(ctx context.Context) (*hcsschema.Properties, error) {
+	// See if we'd even be able to open the silo. We'll need to be local
+	// system/system so check first.
+	usr, err := windows.GetCurrentProcessToken().GetTokenUser()
+	if err != nil {
+		return nil, err
+	}
+	if !usr.User.Sid.IsWellKnown(windows.WinLocalSystemSid) {
+		return nil, errors.New("process does not have the right permissions to open the silo")
+	}
+
+	jobOptions := &jobobject.Options{
```

Comment:
Could we avoid the hcssystem -> job object -> process dance by asking HCS for the process list directly, and iterating over that? Or does that keep us from computing the remaining statistics?

Reply:
So you can issue a ProcessList query to HCS for the PropertiesV2 call, but they end up doing the same NtQuerySystemInformation dance to get the info, although this time at least utilizing more fields. There's no way that I can see to grab just the pids from HCS without having it do all that extra work (filling in the fields in this slice for every pid https://github.com/microsoft/hcsshim/blob/master/internal/hcs/schema2/properties.go#L41). The process list query doesn't return IO information also, to touch on the 'computing the remaining statistics' bit